File: web/datalogui/src/app/classes/topology.ts
import { Dataset, Field } from "./dataset"
import { Tool, ToolAction } from "./toolAction"

export class ProjectFileDir {
  project: string = ""
  name: string = ""
  childDirs?: ProjectFileDir[]
  isFile: boolean = false
  fileContent: string = ""
}

export class Topology {
  tools!: Tool[]
  actions: ToolAction[] = []
  datasets!: Dataset[]
  projectFiles?: ProjectFileDir[]
  fields?: Field[] 
}

File: src/main/scala/ru/neoflex/datalog/domain/Topology.scala
package ru.neoflex.datalog.domain

import ru.neoflex.datalog.engine.dto.Field

case class ProjectFileDir(project: String, name: String,
                          childDirs: List[ProjectFileDir] = List(), isFile: Boolean = false, fileContent: String = "")
case class Topology(tools: List[Tool] = List(), actions: Option[List[ToolAction]], datasets: List[Table] = List(),
                    projectFiles: List[ProjectFileDir] = List(), fields: List[Field] = List())